blockstore/splitstore/markset_badger.go

```diff
@@ -184,8 +186,53 @@ func (s *BadgerMarkSet) Has(c cid.Cid) (bool, error) {
 		return false, nil
 
 	default:
-		return false, xerrors.Errorf("error checking badger markset: %w", err)
+		return false, err
+	}
+}
+
+// writer holds the exclusive lock; put releases it
+func (s *BadgerMarkSet) put(key string) error {
+	s.pend[key] = struct{}{}
+	if len(s.pend) < badgerMarkSetBatchSize {
+		s.mx.Unlock()
+		return nil
+	}
+
+	pend := s.pend
+	seqno := s.seqno
+	s.seqno++
+	s.writing[seqno] = pend
+	s.pend = make(map[string]struct{})
+	s.mx.Unlock()
+
+	defer func() {
+		s.mx.Lock()
+		defer s.mx.Unlock()
+
+		s.version++
+		delete(s.writing, seqno)
+		if len(s.writing) == 0 {
+			s.cond.Broadcast()
```

Comment:
waitgroup?

Reply:
This seems like a case where we're never going to call back into this after the user tries to close, as far as I can tell at least.

Reply:
Not critical, but I prefer to avoid conditions unless I need them as most people don't know about them (let alone understand how to use them correctly).

Reply:
maybe, but this is cleaner imho -- readers should be educated about condition variables, it's an invaluable tool in concurrency.

Reply:
Eh, doesn't really matter and it does mean we only track the count in one place.